### PR TITLE
#9007 Cleanup mods page & remove standalone mod components

### LINKED ISSUE #9007: Drop support for standalone mod components - Slice 3 - Cleanup standalone component code in extension console
See here: https://www.notion.so/pixiebrix/Drop-support-for-standalone-mod-components-f7f3b59fc3474e45955a7b7d46baf333#1e86868c7fad4c45b66620fc30150354

* Refactor/simplify `ModsPage` and the mod view item types/logic to remove handling for standalone mod components
* Clean up/simplify the `useMods` and `useModViewItems` hooks/logic

Follows on from #8979 & #8980

### PR DESCRIPTION
**Pull request by BLoe**
## What does this PR do?

- Closes #9007 

## Demo

- _Paste a screenshot or demo video here_

## Remaining Work

- [x] ~Add new files to strict null checks~
- [ ] Fix/refactor e2e tests

For more information on our expectations for the PR process, see the
[code review principles doc](https://www.notion.so/pixiebrix/Code-Review-Principles-1ce7276b82a84d2a995d55ad85e1310d?pvs=4)

### CODE REVIEW

File: src/extensionConsole/pages/mods/modsPageSelectors.ts [hunk 1]
```diff
@@ -30,3 +46,98 @@ export const selectActiveTab = ({ modsPage }: ModsPageRootState) =>
   modsPage.activeTab;
 export const selectSearchQuery = ({ modsPage }: ModsPageRootState) =>
   modsPage.searchQuery;
+
+const selectActivatedModIds = createSelector(
```
Comment (BLoe) [inline src/extensionConsole/pages/mods/modsPageSelectors.ts:50]: This is the best place to start in order to understand the "progressive data loading" pattern here, where we compose many selectors together in order to break up the former useMods/useModViewItems logic

File: src/extensionConsole/pages/mods/modsPageSelectors.ts [hunk 2]
```diff
@@ -30,3 +46,98 @@ export const selectActiveTab = ({ modsPage }: ModsPageRootState) =>
...
+  selectActivatedModComponents,
+  (activatedModComponents) =>
+    new Set(activatedModComponents.map(({ _recipe }) => _recipe?.id)),
```
Comment (BLoe) [inline src/extensionConsole/pages/mods/modsPageSelectors.ts:53]: FYI - There's an assumption here (and several other places) that `_recipe` will never be undefined, so we don't need to use `compact` or `filter` here to remove undefined items.

File: src/types/modDefinitionTypes.ts [hunk 3]
```diff
@@ -143,3 +143,28 @@ export interface ModDefinition extends UnsavedModDefinition {
...
+    organizations: UUID[];
+  }
+
+  apiVersion: ApiVersion;
+  kind: Kind;
+  extensionPoints: ModComponentDefinition[];
+  definitions?: InnerDefinitions;
+  options?: ModOptionsDefinition;
+
+ */
```
Comment (grahamlangford) [inline src/types/modDefinitionTypes.ts:170]: This intentional?

File: src/modDefinitions/modDefinitionsSlice.ts [hunk 4]
```diff
@@ -52,104 +47,79 @@ export const initialState: ModDefinitionsState = Object.freeze({
...
-      setErrorOnState(state, action.payload);
-      state.isFetchingFromRemote = false;
-      state.isLoadingFromRemote = false;
-    },
-  },
-  extraReducers(builder) {
-    builder.addCase(revertAll, () => initialState);
-  },
-});
-
```
Comment (grahamlangford) [inline src/modDefinitions/modDefinitionsSlice.ts:100]: Love this. We should validate all of our slices and be sure we're using the extraReducers correctly

File: src/modDefinitions/modDefinitionsListenerMiddleware.ts [hunk 5]
```diff
@@ -30,7 +30,7 @@ modDefinitionsListenerMiddleware.startListening({
     apiEndpoints.updatePackage.matchFulfilled,
     apiEndpoints.deletePackage.matchFulfilled,
   ),
-  effect(action, { dispatch }) {
+  effect(action, { dispatch, getState }) {
```
Comment (grahamlangford) [inline src/modDefinitions/modDefinitionsListenerMiddleware.ts:33]: unused getState

Comment (grahamlangford) [inline src/modDefinitions/modDefinitionsListenerMiddleware.ts:33]:
```suggestion
  effect(action, { dispatch }) {
```

File: src/types/modDefinitionTypes.ts (hunk 3 above)
Comment (BLoe) [reply src/types/modDefinitionTypes.ts:170]: oops

File: src/modDefinitions/modDefinitionsSlice.ts (hunk 4 above)
Comment (BLoe) [reply src/modDefinitions/modDefinitionsSlice.ts:100]: I have more refactoring in mind at some point for this slice, but the mis-use of thunks was the most egregious thing here that I wanted to fix/clean up for now.

File: src/modDefinitions/modDefinitionsListenerMiddleware.ts (hunk 5 above)
Comment (BLoe) [reply src/modDefinitions/modDefinitionsListenerMiddleware.ts:33]: Whoops, I was trying something here briefly, forgot to remove

File: src/extensionConsole/pages/mods/modsPageSelectors.ts (hunk 2 above)
Comment (BLoe) [reply src/extensionConsole/pages/mods/modsPageSelectors.ts:53]: Ok will do

File: src/extensionConsole/pages/mods/utils/buildGetModVersionStatus.ts [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { type ActivatedModComponent } from "@/types/modComponentTypes";
+import { type Mod, type ModVersionStatus } from "@/types/modTypes";
+import { assertNotNullish } from "@/utils/nullishUtils";
+import { isUnavailableMod } from "@/utils/modUtils";
+import * as semver from "semver";
+
+export default function buildGetModVersionStatus(
```
Comment (twschiller) [inline src/extensionConsole/pages/mods/utils/buildGetModVersionStatus.ts:24]: FWIW, these can just be defined as normal function taking the activated mod components and the mod. And then at the callsite use lodash curry or just plain JS

File: src/components/MarketplaceListingIcon.tsx [hunk 7]
```diff
@@ -38,7 +38,8 @@ const MarketplaceListingIcon: React.FC<{
   faIconClass = "",
   inheritColor = false,
 }) => {
-  const { data: listing } = useGetMarketplaceListingQuery({ packageId });
+  const queryInput = useMemo(() => ({ packageId }), [packageId]);
+  const { data: listing } = useGetMarketplaceListingQuery(queryInput);
```
Comment (grahamlangford) [inline src/components/MarketplaceListingIcon.tsx:42]: Interesting. Did you validate that the useMemo actually has a positive performance impact? I thought RTK Query stringified the queryInput for caching, so memoization wouldn't help?

File: src/store/modComponents/modComponentSelectors.ts [hunk 8]
```diff
@@ -46,22 +46,17 @@ export const selectIsModComponentSavedOnCloud =
...
-  (modId: RegistryId) => (state: ModComponentsRootState) =>
-    activatedModComponentsForModSelector(state, modId);
-
 export const selectModHasAnyActivatedModComponents =
   (modId?: RegistryId) =>
   (state: ModComponentsRootState): boolean =>
-    Boolean(
-      modId && !isEmpty(activatedModComponentsForModSelector(state, modId)),
-    );
+    Boolean(modId && !isEmpty(selectGetModComponentsForMod(state)(modId)));
```
Comment (grahamlangford) [inline src/store/modComponents/modComponentSelectors.ts:62]: NIT: should we go ahead and convert this to use reselect?

File: src/components/MarketplaceListingIcon.tsx (hunk 7 above)
Comment (BLoe) [reply src/components/MarketplaceListingIcon.tsx:42]: Hmm yeah this doesn't seem to have any effect when I check how often the listing is updated 🤔 🤷‍♂️  I'll revert. I think I was fixing something and tried this and then figured it out another way and didn't go back and clean this up.

File: src/store/modComponents/modComponentSelectors.ts (hunk 8 above)
Comment (BLoe) [reply src/store/modComponents/modComponentSelectors.ts:62]: It's only called once with the modId on the activate mod page, so I don't think there are really any perf concerns here right now

File: src/extensionConsole/pages/mods/gridView/GridView.tsx [hunk 9]
```diff
@@ -112,12 +111,9 @@ const GridView: React.VoidFunctionComponent<ModsPageContentProps> = ({
     [expandedGridRows],
   );
 
-  // `react-window` caches itemSize which causes inconsistent
-  // row heights/row height flickering on scroll when data changes,
-  // even with non-index `itemKeys`.
   // Re-render the list when expandedRows changes.
   useEffect(() => {
-    setListKey(uuidv4());
+    listRef.current?.resetAfterIndex(0);
```
Comment (grahamlangford) [inline src/extensionConsole/pages/mods/gridView/GridView.tsx:116]: Nice!

**Review by grahamlangford — APPROVED** [review]

**Comment by twschiller** [comment]
@BLoe see conflicts - @grahamlangford's PR renamed some things

**Pull request merged**